sky/provision/gcp/instance_utils.py

```diff
@@ -706,29 +687,110 @@ def create_instances(
+            # by creating VMs with machine images on different zones.
+            operation = request.execute(num_retries=GCP_MAX_RETRIES)
+            operations.append(operation)
+
+        logger.debug('"insert" operation requested ...')
+        return operations
+
+    @classmethod
+    def _bulk_insert(cls, names: List[str], project_id: str, zone: str,
+                     config: dict) -> List[dict]:
+        source_instance_template = config.pop('sourceInstanceTemplate', None)
+        if 'scheduling' in config and isinstance(config['scheduling'], list):
+            # For backeward compatibility: converting the list of dictionaries
+            # to a dictionary due to the use of deprecated API.
+            # [{'preemptible': True}, {'onHostMaintenance': 'TERMINATE'}]
+            # to {'preemptible': True, 'onHostMaintenance': 'TERMINATE'}
+            config['scheduling'] = {
+                k: v for d in config['scheduling'] for k, v in d.items()
+            }
+
+        for disk in config.get('disks', []):
+            disk_type = disk.get('initializeParams', {}).get('diskType')
+            if disk_type:
+                disk['initializeParams']['diskType'] = selflink_to_name(
+                    disk_type)
+        config['machineType'] = selflink_to_name(config['machineType'])
+        for accelerator in config.get('guestAccelerators', []):
+            accelerator['acceleratorType'] = selflink_to_name(
+                accelerator['acceleratorType'])
+
+        body = {
+            'count': len(names),
+            'instanceProperties': config,
+            'sourceInstanceTemplate': source_instance_template,
+            'perInstanceProperties': {n: {} for n in names}
+        }
+        logger.debug('Launching GCP instances with "bulkInsert" ...')
+        request = cls.load_resource().instances().bulkInsert(
+            project=project_id,
+            zone=zone,
+            body=body,
+        )
+        operation = request.execute(num_retries=0)
+        return [operation]
+
+    @classmethod
+    def _use_bulk_insert(cls, config) -> bool:
+        """Decide whether to use bulkInsert or not based on config."""
+        # bulkInsert does not support overriding parameter sourceMachineImage
+        # with disks (without a sourceImage), causing the following error:
+        #   'Invalid value for field \'resource.instanceProperties.disks[0]\':
+        #   \'{  "type": "PERSISTENT",  "boot": true,  "initializeParams": {
+        #    "diskSizeGb": "256",    "diskType"...\'. Boot disk must have a
+        #    source specified'
+        # https://cloud.google.com/compute/docs/reference/rest/v1/instances/bulkInsert # pylint: disable=line-too-long
+        if config.get('sourceMachineImage'):
+            return False
+        return True
+
     @classmethod
     def _create_instances(
         cls,
         names: List[str],
         project_id: str,
         zone: str,
         config: dict,
-        count: int,
```

Comment:
Why remove this?

Reply:
This is duplicated with `len(names)`. We should be able to leave the `count` out.